### PR TITLE
I <3 trains

### PR DESCRIPTION
**Pull request by marto55**

### CODE REVIEW

File: homework_02/Martin_Georgiev_13/views.py [hunk 1]
```diff
@@ -5,19 +5,67 @@
...
-    response = requests.get("https://api-v3.mbta.com/predictions?page%5Boffset%5D=0&page%5Blimit%5D=10&sort=-departure_time&filter%5Bstop%5D=place-north").json()
+    params = {
+        "sort" : "-arrival_time",
+        "filter[direction_id]" : "0",
+        "page[offset]" : "0",
+        "include" : "trip,schedule",
+        "page[limit]" : "100",
+        "filter[route_type]" : "2",
+        "filter[stop]" : "place-north",
+        "filter[direction_id]" : "0",
```
Comment (Teodor1331) [inline homework_02/Martin_Georgiev_13/views.py:16]: Okay, you have taken only the trains that will departure, so, perfect here!

File: homework_02/Martin_Georgiev_13/views.py [hunk 2]
```diff
@@ -5,19 +5,67 @@
...
+    current_time = datetime.datetime.now(pytz.timezone("America/New_York"))
+    response = requests.get("https://api-v3.mbta.com/predictions", params = params).json()
+
+    destination_times = []
+    trip_ids = []
+    directions = []
     statuses = []
-    for status in response['data']:
-        statuses.append(status['attributes']['status'])
+    train_ids = []
```
Comment (Teodor1331) [inline homework_02/Martin_Georgiev_13/views.py]: Cool, you have splitted the different fields of the information in different lists!

File: homework_02/Martin_Georgiev_13/views.py [hunk 3]
```diff
@@ -5,19 +5,67 @@
...
+        "month" : current_time.month,
+        "time" : current_time.strftime("%H:%M:%S"),
+        "time" : current_time.strftime("%I:%M %p"),
+        "trips" : trip_ids,
+        "destination_times" : destination_times,
         "statuses" : statuses,
-        'data': response,
+        "directions" : directions,
+        "train_ids" : train_ids,
+        "data": response,
```
Comment (Teodor1331) [inline homework_02/Martin_Georgiev_13/views.py]: The new dictionary seems better and I personally like it!

File: homework_02/Martin_Georgiev_13/views.py [hunk 4]
```diff
@@ -5,19 +5,67 @@
...
+                    train_ids.append(train["attributes"]["name"])
+
+    for direction in response["included"]:
+        if direction["type"] == "trip":
+            for id in trip_ids:
+                if direction["id"] == id:
+                    directions.append(direction["attributes"]["headsign"])
+
+    for status in response["data"]:
+        statuses.append(status["attributes"]["status"])
```
Comment (Teodor1331) [inline homework_02/Martin_Georgiev_13/views.py:55]: The algorithm seems perfect for me. You have taken the right attributes. You can simplify some of the loops with lambda expressions, but I think here is okay without lambda expressions, too.

File: homework_02/Martin_Georgiev_13/Templates/trains.html [hunk 5]
```diff
@@ -17,43 +17,57 @@
...
-            <div style= "float: center;">
-    
-                <h1>Current time : {{ cur_date }}-{{ cur_month }}-{{ cur_year }} | {{ cur_time }}</h1>
-            </div>
+<body>
+    <br>
+    <div id="time">
+        <div style= "float: center;">
+
+            <h1>Current time : {{ day }}-{{ month }}-{{ year }} | {{ time }}</h1>
```
Comment (Teodor1331) [inline homework_02/Martin_Georgiev_13/Templates/trains.html:27]: The format here is good. You have maintained the identation, so, I have no objections here.

File: homework_02/Martin_Georgiev_13/views.py [hunk 6]
```diff
@@ -5,19 +5,67 @@
...
     context = {
-        "cur_year" : cur_time.year,
-        "cur_date" : cur_time.day,
-        "cur_month" : cur_time.month,
-        "cur_time" : cur_time.strftime("%H:%M:%S"),
+        "year" : current_time.year,
+        "day" : current_time.day,
+        "month" : current_time.month,
+        "time" : current_time.strftime("%H:%M:%S"),
+        "time" : current_time.strftime("%I:%M %p"),
```
Comment (Teodor1331) [inline homework_02/Martin_Georgiev_13/views.py:62]: Avoid in the future to have same key for different values in a dictionary.

File: homework_02/Martin_Georgiev_13/Templates/trains.html [hunk 7]
```diff
@@ -17,43 +17,57 @@
...
+                    {% if forloop.counter == forloop.parentloop.counter %}
+                        <th><h4>{{ id }}</h4></th>
+                    {% endif %}
+                {% endfor %}
+
+                {% for status in statuses %}
+                    {% if forloop.counter == forloop.parentloop.counter %}
+                        <th><h4>{{ status }}</h4></th>
+                    {% endif %}
+                {% endfor %}
```
Comment (Teodor1331) [inline homework_02/Martin_Georgiev_13/Templates/trains.html:65]: The presentation of the attributes is cool.

**Review by Teodor1331 — APPROVED** [review]

**Review by Vic-Dim — COMMENTED** [review]
Well done.

**Pull request merged**